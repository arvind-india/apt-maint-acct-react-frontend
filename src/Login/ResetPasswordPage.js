import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import {
          Form,
          Button,
          FormGroup,
          FormText,
          Input,
          Label,
          Col
        } from 'reactstrap'

import { userActions, alertActions } from '../_actions'
import { FlashMessage } from '../_components'

export class ResetPassword extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      password: '',
      confirmPassword: '',
      submitted: false
    }
    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }
  handleChange(event) {
    const { name, value } = event.target
    this.setState({
        [name]: value
    })
  }
  handleSubmit(event) {
    event.preventDefault()
    this.setState({ submitted: true })
    const { password, confirmPassword } = this.state
    const { match } = this.props
    if(password &&
        confirmPassword &&
        match.params.token &&
        password === confirmPassword
    ) {
      this.props.resetPassword(password, match.params.token)
    } else {
      this.props.error('Missing Date...')
    }
  }
  render() {
    const { alert } = this.props
    return (
      <div>
        <h2 align="center">Reset Password</h2>
        {alert.message && <FlashMessage text={alert.message} color={alert.color} delay={2100}/>}
        <Form id="resetPasswordForm" onSubmit={this.handleSubmit}>
          { this.password() }
          { this.repeatPassword() }
          { this.buttons() }
        </Form>
      </div>
    )
  }
  password() {
    const { submitted, password } = this.state
    return <FormGroup row>
      <Label sm={3}>Password</Label>
      <Col sm={9}>
        <Input
          id="password"
          type="password"
          name="password"
          placeholder="password here"
          onChange={this.handleChange}
        />
        {submitted && !password && <FormText color="danger">Password is required</FormText>}
      </Col>
    </FormGroup>
  }
  repeatPassword() {
    const { submitted, password, confirmPassword } = this.state
    return <FormGroup row>
      <Label sm={3}>Repeat_Password</Label>
      <Col sm={9}>
        <Input
          id="confirmPassword"
          type="password"
          name="confirmPassword"
          placeholder="Confirm password here"
          onChange={this.handleChange}
        />
        {submitted && !confirmPassword && <FormText color="danger">Repeat Password is required</FormText>}
        {password && confirmPassword && password !== confirmPassword && <FormText color="danger">Password do NOT matchG</FormText>}
      </Col>
    </FormGroup>
  }

  buttons() {
    return <FormGroup row>
      <Label sm={3}></Label>
      <Col smoffset={3} sm={9}>
        <Button type="submit" color="primary" title="Reset Password">Reset</Button>
        <Button color="link">
          <Link to="/login" className="text-danger" title="Go to home">Cancel</Link>
        </Button>
      </Col>
    </FormGroup>
  }

}

function mapDispatchToProps(dispatch) {
  return {
    resetPassword: (password, token) => {
      dispatch(userActions.resetPassword(password, token))
    },
    error: (msg) => {
      dispatch(alertActions.error(msg))
    }
  }
}

function mapStateToProps(state) {
  const { alert } = state
  return {
    alert
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(ResetPassword)
