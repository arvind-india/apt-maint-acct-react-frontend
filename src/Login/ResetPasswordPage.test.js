// ResetPasswordPage.test.js

import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer';

import { ResetPassword } from './ResetPasswordPage'

const mockProps = {
  alert: {message: "testingAlert"},
  // authentication: {loggingIn: true, user: {}},
  // login: jest.fn(),  // mock function
  // logout: jest.fn(), // mock function
  // clearAlert: jest.fn(), // mock function
  resetPassword: jest.fn(),
  error: jest.fn(),
  match: { params: { token: '1234'}}
}
const mockStates = {
  // email: 'unitTesting@jest.js',
  password: 'secret',
  confirmPassword: 'secret'
}
/* const mockEmail = {
  name: 'email',
  value: "dummy@email.id"
} */

const mockPassword = {
  name: 'password',
  value: "secret"
}
const mockRepeatPassword = {
  name: 'confirmPassword',
  value: "secret"
}

const mockResult = [
//  mockEmail.value,
  mockPassword.value,
  mockProps.match.params.token
]
const component = shallow(<ResetPassword {...mockProps} />)

describe('ResetPasswordPage params test', () => {
  it('should render ResetPasswordPage without crashing', () => {
    component
  })
  it('should render initial layout', () => {
    expect(component.getElements()).toMatchSnapshot()
  })
/*  it('should have email state', () => {
    expect(component.state('email')).toBeDefined()
  }) */
  it('should have password state', () => {
    expect(component.state('password')).toBeDefined()
  })
  it('should have repeat password state', () => {
    expect(component.state('confirmPassword')).toBeDefined()
  })
/*  it('should set email state', () => {
    component.setState({email: mockStates.email})
    expect(component.state('email')).toEqual(mockStates.email)
  }) */
  it('should set password state', () => {
    component.setState({password: mockStates.password})
    expect(component.state('password')).toEqual(mockStates.password)
  })
  it('should set repeat password state', () => {
    component.setState({confirmPassword: mockStates.confirmPassword})
    expect(component.state('confirmPassword')).toEqual(mockStates.confirmPassword)
  })
})

describe('ResetPasswordPage events test', () => {
  it('should call mock login function', () => {
    component.find('#resetPasswordForm').simulate('submit', {preventDefault() {}})
    expect(mockProps.resetPassword.mock.calls.length).toEqual(1)
  })
  it('should be called with the password and confirm password in the state as arguments', () => {
    // component.find('#email').simulate('change', {target: mockEmail})
    component.find('#password').simulate('change', {target: mockPassword})
    component.find('#confirmPassword').simulate('change', {target: mockRepeatPassword})
    component.find('#resetPasswordForm').simulate('submit', {preventDefault() {}})
    expect(mockProps.resetPassword.mock.calls[1]).toEqual(mockResult)
  })
})
