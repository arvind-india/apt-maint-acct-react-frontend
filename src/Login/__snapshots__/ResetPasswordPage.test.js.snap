// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPasswordPage params test should render initial layout 1`] = `
Array [
  <div>
    <h2
        align="center"
    >
        Reset Password
    </h2>
    <FlashMessage
        color={undefined}
        delay={4000}
        text="testingAlert"
    />
    <Form
        id="resetPasswordForm"
        onSubmit={[Function]}
        tag="form"
    >
        <FormGroup
            row={true}
            tag="div"
        >
            <Label
                sm={3}
                tag="label"
                widths={
                    Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                }
            >
                Password
            </Label>
            <Col
                sm={9}
                tag="div"
                widths={
                    Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                }
            >
                <Input
                    id="password"
                    name="password"
                    onChange={[Function]}
                    placeholder="password here"
                    tag="p"
                    type="password"
                />
            </Col>
        </FormGroup>
        <FormGroup
            row={true}
            tag="div"
        >
            <Label
                sm={3}
                tag="label"
                widths={
                    Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                }
            >
                Repeat_Password
            </Label>
            <Col
                sm={9}
                tag="div"
                widths={
                    Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                }
            >
                <Input
                    id="confirmPassword"
                    name="confirmPassword"
                    onChange={[Function]}
                    placeholder="Confirm password here"
                    tag="p"
                    type="password"
                />
                
            </Col>
        </FormGroup>
        <FormGroup
            row={true}
            tag="div"
        >
            <Label
                sm={3}
                tag="label"
                widths={
                    Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                }
            />
            <Col
                sm={9}
                smoffset={3}
                tag="div"
                widths={
                    Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                }
            >
                <Button
                    color="primary"
                    tag="button"
                    title="Reset Password"
                    type="submit"
                >
                    Reset
                </Button>
                <Button
                    color="link"
                    tag="button"
                >
                    <Link
                        className="text-danger"
                        replace={false}
                        title="Go to home"
                        to="/login"
                    >
                        Cancel
                    </Link>
                </Button>
            </Col>
        </FormGroup>
    </Form>
</div>,
]
`;
